### PR TITLE
Verify environment presence in the useEnvironment middleware

### PR DESCRIPTION
**Pull request by 2can**
- Move verification logic from `env remove` to useEnvironment middleware
- Allow to pass environment option/positional by name
- Introduce spinner for better UX
- Show custom error message for missing env

### CODE REVIEW

File: src/cli/env/remove.ts [hunk 1]
```diff
@@ -23,36 +23,13 @@ export const builder: CommandBuilder = (_) =>
...
-    `environment ${env.name} - ${env.key}`
-  );
+  const proceed = await confirmRemoval(argv, `environment ${environment}`);
 
   if (proceed && environment) {
-    const result = (await DELETE(API.Environment, argv)) as any;
+    const result = (await DELETE(API.Environment, {
+      ...argv,
+      ...{ environment },
+    })) as any;
```
Comment (lkostrowski) [inline src/cli/env/remove.ts:32]: Please avoid `any` casting. It basically disables typescript.

I suggest `as unknown` and prepare a type narrowing check - something like `typeof result === 'object' && 'task_id' in result`

File: src/middleware/index.ts [hunk 2]
```diff
@@ -86,33 +87,117 @@ export const useOrganization = async ({ token, organization }: Options) => {
...
+          );
+          process.exit(1);
+        }
+
+        throw error;
+      }
+
+      throw error;
+    }
+  }
```
Comment (lkostrowski) [inline src/middleware/index.ts:142]: This block has quite high complexity. It introduces `any` and nests both errors and conditionals.

Could you refactor it, maybe extracting some methods?

File: src/middleware/index.ts [hunk 3]
```diff
@@ -86,33 +87,117 @@ export const useOrganization = async ({ token, organization }: Options) => {
...
 
+const verifyEnvironment = async (
+  token: string,
+  organization: string,
+  environment: string
+) => {
+  const spinner = ora(
+    chalk(chalk.bold('Environment ·'), chalk.cyan(environment))
+  ).start();
+  let env;
```
Comment (lkostrowski) [inline src/middleware/index.ts:98]: `let env` is hidden `any` type. We should avoid that

**Review by lkostrowski — CHANGES_REQUESTED** [review]

**Review by zaiste — APPROVED** [review]

**Comment by lkostrowski** [comment]
@michalina-graczyk can you please test it?

**Pull request merged**